packages/peregrine/lib/talons/Header/useAccountMenu.js

```diff
@@ -58,14 +58,12 @@ export const useAccountMenu = props => {
         setAccountMenuIsOpen(false);
     }, [location.pathname, setAccountMenuIsOpen]);
 
-    // Set view to sign in / account when the dropdown is closed
+    // Update view based on user status everytime accountMenuIsOpen has changed.
     useEffect(() => {
-        if (!accountMenuIsOpen) {
-            if (isUserSignedIn) {
-                setView('ACCOUNT');
-            } else {
-                setView('SIGNIN');
-            }
+        if (isUserSignedIn) {
```

Comment:
Instead of writing 2 if cases, 1 for `accountMenuIsOpen` is true and 1 for false, i took the approach of adding `accountMenuIsOpen` to dependency array even though it is not used in the effect. This works but a new approach we haven't seen before in our code base. Let me know what you guys think.

Reply:
We do this in the affect immediately above this with `location.pathname`. And you've already left a comment to that affect above here so I'm fine with it.